SUIApp/SUIApp/ViewModel/ClimateViewModel.swift

```diff
@@ -0,0 +1,80 @@
+// ClimateViewModel.swift
+// Copyright © RoadMap. All rights reserved.
+
+import SwiftUI
+
+/// ViewModel для экрана настройки климата
+final class ClimateViewModel: ObservableObject {
+    // MARK: - Private Constants
+
+    private enum Constants {
+        static let degreeText = "° C"
+        static let emptyText = ""
+        static let acOnText = "A/C is ON"
+        static let acOffText = "A/C is OFF"
+        static let topGradientColorName = "topGradient"
+        static let bottomGradientColorName = "bottomGradient"
+    }
+
+    // MARK: - Public Property
+
+    @Published var isStart = false
+    @Published var isAlertShown = false
+    @Published var acSlider = 15.0
+    @Published var fanSlider = 0.0
+    @Published var heatSlider = 0.0
+    @Published var autoSlider = 0.0
+    @Published var currentColor = Color(Constants.bottomGradientColorName)
+    @Published var currentMenuOffsetY: CGFloat = -200
+    @Published var lastMenuOffsetY: CGFloat = 0
+
+    var updateSlider: Double {
+        (acSlider - 15.0) / 15.0
+    }
+
+    var updateTemperature: String {
+        isStart ? "\(Int(acSlider)) \(Constants.degreeText)" : Constants.emptyText
+    }
+
+    var updateClimate: String {
+        isStart ? "\(Int(acSlider)) \(Constants.degreeText)" : Constants.emptyText
+    }
+
+    var updateACTitle: String {
+        isStart ? Constants.acOnText : Constants.acOffText
+    }
+
+    var updateSliderColor: Color {
+        isStart ? Color(Constants.topGradientColorName) : .gray
+    }
+
+    // MARK: - Public Methods
+
+    func incrementAC() {
+        acSlider += acSlider < 30 ? 1 : 0
+    }
+
+    func decrementAC() {
+        acSlider -= acSlider > 15 ? 1 : 0
+    }
+
+    func resetSetting() {
+        withAnimation {
+            isStart.toggle()
+            acSlider = isStart ? acSlider : 15.0
+        }
+    }
+
+    func updateOffset(max: CGFloat) {
+        if -currentMenuOffsetY > max / 3 {
+            currentMenuOffsetY = -max / 2.2
+        } else {
+            currentMenuOffsetY = -200
```

Comment:
```suggestion
            currentMenuOffsetY = -200
```
2.2, 3, 2.2 , -200 маг число